x-pack/plugins/fleet/public/applications/fleet/sections/agents/agent_details_page/components/agent_details/agent_details_integration.tsx

```diff
@@ -0,0 +1,211 @@
+import { useLink, useUIExtension } from '../../../../../hooks';
+import { ExtensionWrapper, PackageIcon } from '../../../../../components';
+
+import { AgentDetailsIntegrationInputs } from './agent_details_integration_inputs';
+import { getInputStatusFromAgent } from './input_status_utils';
+
+const StyledEuiAccordion = styled(EuiAccordion)`
+  .euiAccordion__button {
+    width: 90%;
+  }
+
+  .euiAccordion__triggerWrapper {
+    padding-left: ${(props) => props.theme.eui.euiSizeM};
+  }
+
+  &.euiAccordion-isOpen {
+    .euiAccordion__childWrapper {
+      padding: ${(props) => props.theme.eui.euiSizeM};
+      padding-top: 0px;
+    }
+  }
+
+  .ingest-integration-title-button {
+    padding: ${(props) => props.theme.eui.euiSizeS};
+  }
+
+  .euiTableRow:last-child .euiTableRowCell {
+    border-bottom: none;
+  }
+
+  .euiIEFlexWrapFix {
+    min-width: 0;
+  }
+
+  .euiAccordion__buttonContent {
+    width: 100%;
+  }
+`;
+
+const CollapsablePanel: React.FC<{
+  id: string;
+  title: React.ReactNode;
+  'data-test-subj'?: string;
+}> = ({ id, title, children, 'data-test-subj': dataTestSubj }) => {
+  const arrowProps = useMemo<EuiAccordionProps['arrowProps']>(() => {
+    if (dataTestSubj) {
+      return {
+        'data-test-subj': `${dataTestSubj}-openCloseToggle`,
+      };
+    }
+    return undefined;
+  }, [dataTestSubj]);
+
+  return (
+    <EuiPanel paddingSize="none">
+      <StyledEuiAccordion
+        id={id}
+        arrowDisplay="left"
+        buttonClassName="ingest-integration-title-button"
+        buttonContent={title}
+        arrowProps={arrowProps}
+        data-test-subj={dataTestSubj}
+      >
+        {children}
+      </StyledEuiAccordion>
+    </EuiPanel>
+  );
+};
+
+export const AgentDetailsIntegration: React.FunctionComponent<{
+  agent: Agent;
+  agentPolicy: AgentPolicy;
+  packagePolicy: PackagePolicy;
+  'data-test-subj'?: string;
+}> = memo(({ agent, agentPolicy, packagePolicy, 'data-test-subj': dataTestSubj }) => {
+  const { getHref } = useLink();
+  const theme = useEuiTheme();
+
+  const [isAttentionBadgeNeededForPolicyResponse, setIsAttentionBadgeNeededForPolicyResponse] =
+    useState(false);
+
+  const policyResponseExtensionView = useUIExtension(
+    packagePolicy.package?.name ?? '',
+    'package-policy-response'
+  );
+
+  const policyResponseExtensionViewWrapper = useMemo(() => {
+    return (
+      policyResponseExtensionView && (
+        <ExtensionWrapper>
+          <policyResponseExtensionView.Component
+            agent={agent}
+            onShowNeedsAttentionBadge={setIsAttentionBadgeNeededForPolicyResponse}
+          />
+        </ExtensionWrapper>
+      )
+    );
+  }, [agent, policyResponseExtensionView]);
+
+  const packageErrors = useMemo(() => {
+    const packageErrorUnits: FleetServerAgentComponentUnit[] = [];
+    if (!agent.components) {
+      return packageErrorUnits;
+    }
+    return getInputStatusFromAgent(agent.components, packagePolicy).filter(
```

Comment:
suggestion for a shorter way with [lodash](https://lodash.com/docs/4.17.15#some):
```
hasPackageErrors = agent.components ? some(getInputStatusFromAgent(agent.components, packagePolicy), unit => unit.status === 'DEGRADED' || unit.status === 'FAILED') : false;
```

Reply:
Thanks for this @juliaElastic - there actually was some leftover unnecessary code there which I've removed.